torch/csrc/jit/fuser/compiler.cpp

```diff
@@ -304,15 +315,29 @@ std::shared_ptr<FusedKernel> compileKernel(
     }
   }
 
+  // if SumToSize changes the number of dimensions of FusedConcat
+  // inputs, we need to adjust the concat dimension.
+  // here we compute the needed dimension.
+  std::vector<int64_t> dims_to_go(graph->outputs().size(), 0);
+  for (const auto& omap : spec.outputMapAndSizes()) {
+    auto size_inputs = omap.sizeInputs();
+    if (size_inputs.size() > 0 && size_inputs[0] != -1) {
+      dims_to_go[omap.offset()] =
+          map_size.size() - all_inputs[size_inputs[0]].toIntListRef().size();
```

Comment:
You can't adjust the concat dimension only on the basis of how many dimensions will be removed. The important part is how many dimensions to the left of the specified dimension (in case of a positive index), or to the right (in case of a negative index).

Reply:
For negative indices, we would have no change, I'll have to double-check whether they happen here.
For positive indices, the dimensions that are removed necessarily lie on the left, because otherwise there will be no change.
Note that the removing of dimensions happens here because we're undoing adding dimensions in broadcasting.